packages/aws-cdk-lib/aws-route53/lib/record-set.ts

```diff
@@ -233,6 +243,46 @@ export interface RecordSetOptions {
   readonly setIdentifier?: string;
 }
 
+/**
+ * Configuration for IP-based routing.
+ */
+export interface ICidrRoutingConfig {
+  /**
+   * List of CIDR blocks.
+   *
+   * When specifying the default location (locationName is '*'), it cannot be set, but for all other cases, it is mandatory.
+   *
+   * @default - zero bit CIDR block (0.0.0.0/0 or ::/0) for the default location
+   */
+  cidrList?: string[];
+  /**
+   * The name of the location.
+   *
+   * When '*' is specified, it is treated as the default location.
+   * In this case, the cidrList cannot be specified.
+   */
+  locationName: string;
+  /**
+   * The name of a CIDR collection.
+   *
+   * This parameter is ignored when the `collection` property is specified.
+   *
+   * @default - Auto generated name
+   */
+  collectionName?: string;
+  /**
+   * Existing Cidr Collection.
+   *
+   * Use this to add a new Location to an existing Cidr Collection.
+   * Note that for IP-based routing, all resource record sets for the same record set name and type must reference the same CIDR collection.
+   *
+   * @see https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/routing-policy-ipbased.html
+   *
+   * @default - Create a new CIDR Collection
+   */
+  collection?: CfnCidrCollection;
```

Comment:
Same comment as above. The peculiarities of how these four props interact should be enforced in the contract.